### PR TITLE
feat: new sceneComponent option

### PR DESCRIPTION
**Pull request by wcandillon**
Please provide enough information so that others can review your pull request:

**Motivation**

This allows users to customize drawer animation by providing the ability to set a custom wrapper to the scene.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-navigation-drawer",
-  "version": "2.3.4",
+  "version": "2.4.0",
```
Comment (satya164) [inline package.json]: Our versioning is handled by the release process. So revert this change

**Review by satya164 — CHANGES_REQUESTED** [review]
Thanks for the PR. I think I'd prefer to expose the animated progress value to the screen via context instead of `sceneComponent` prop. We expose similar animation context for stack: https://github.com/react-navigation/react-navigation/blob/424c9469e442d15b31c36ee68d6d98fb419ff9dc/packages/stack/src/index.tsx#L44

**Comment by wcandillon** [comment]
Exposing the progress animation node via the context would be great but we would still need to wrap manually all of our screens with a container that can use this animation node to perform some animation. What are your thoughts on this? Thank you for looking into it.

**Comment by satya164** [comment]
I think it's ok, you can create a HOC/wrapper to reduce the repetition. Wrapping screens in a custom view isn't something a lib should do imo. It makes it easier to break things and may also conflict with any animation options that we want to add in future.

PS: this repo is being archived. Please send a PR here https://github.com/react-navigation/react-navigation/tree/4.x/packages/drawer

**Pull request closed without merging**